control/controlclient/client.go

```diff
@@ -54,6 +54,10 @@ type Client interface {
 	// TODO: It might be better to simply shutdown the controlclient and
 	// make a new one when it's time to unpause.
 	SetPaused(bool)
+	// SetSleepMode pauses the control client and prevents anybody else
+	// from unpausing it until SetSleepMode(false) is called again
+	SetSleepMode(bool)
+	IsSleeping() bool
```

Comment:
Sleep vs Sleeping is weird here (different verb forms for the getter vs setter)

SetSleeping, IsSleeping?